e2e/deployer.go

```diff
@@ -0,0 +1,105 @@
+package e2e
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"strings"
+
+	"k8s.io/kubernetes/test/e2e/framework"
+	e2elog "k8s.io/kubernetes/test/e2e/framework/log"
+)
+
+// deployer interface modify the templates on demand and also help
+// to create or delete templates.
+type deployer interface {
+	replaceNamespaceInTemplate(string) (string, error)
+	runAction(string, string)
+	runActionRookDelete(string, string)
+	changeTemplateAndRunAction(string, string)
+}
+
+type deploy struct{}
+
+func (d deploy) replaceNamespaceInTemplate(filePath string) (string, error) {
+	read, err := ioutil.ReadFile(filePath)
+	if err != nil {
+		return "", err
+	}
+	return strings.ReplaceAll(string(read), "namespace: default", fmt.Sprintf("namespace: %s", cephCSINamespace)), nil
+}
+
+func (d deploy) runAction(template, action string) {
+	data, err := d.replaceNamespaceInTemplate(template)
+	if err != nil {
+		e2elog.Failf("failed to read content from %s with error %v", template, err)
+	}
+	_, err = framework.RunKubectlInput(cephCSINamespace, data, ns, action, "-f", "-")
+	if err != nil {
+		e2elog.Failf("failed to %s %q with error %v", action, template, err)
+	}
+}
+
+func (d deploy) runActionRookDelete(template, action string) {
+	data, err := d.replaceNamespaceInTemplate(template)
+	if err != nil {
+		e2elog.Failf("failed to read content from %s with error %v", template, err)
+	}
+	_, err = framework.RunKubectlInput(cephCSINamespace, data, "--ignore-not-found=true", ns, action, "-f", "-")
+	if err != nil {
+		e2elog.Failf("failed to %s %q with error %v", action, template, err)
+	}
+}
+
+func (d deploy) changeTemplateAndRunAction(template, action string) {
+	switch template {
+	case rbdDirPath + csiDriverObject, cephfsDirPath + csiDriverObject:
+		csiDriver, err := ioutil.ReadFile(rbdDirPath + csiDriverObject)
```

Comment:
for cephfsDir also its reading rbdDirPath?

Reply:
Thanks.. it should have been template,, will correct it.